attribution_scopes.md

```diff
@@ -0,0 +1,236 @@
+  - [Deletion logic](#deletion-logic)
+- [Attribution scope example](#attribution-scope-examples)
+- [Privacy Considerations](#privacy-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+<br>
+The current attribution logic in the Attribution Reporting API may not be ideal for use-cases where an API caller needs more fine grain control over the attribution granularity (i.e. campaign, product, conversion ID, etc.) before a source is chosen for attribution. Currently available features such as top-level filters are not sufficient for this use-case because they happen after a source has been selected (i.e. after destination matching) which results in either no attribution occuring or incorrect attribution depending on the top-level filters that are set. We can support this use-case by allowing registrations to specify predefined attribution scopes that will be considered for filtering *before* attributing a source, in order to more efficiently extract utility out of the API. 
+
+## Goals
+
+In general, the approach here is to allow API callers to specify a list of strings for sources and triggers during registration that can be used for filtering before attribution takes place:
+
+* Allow finer-grained filtering before attribution, trading off additional filtering with noise levels
+
+
+## Pre-Attribution Filtering
+
+### API changes
+
+The following optional parameters will be added to the JSON in  `Attribution-Reporting-Register-Source` during source registration: `attribution_scopes`, `attribution_scope_limit` and `max_event_states`:
+
+```jsonc
+{
+  ..., //existing fields
+
+  // Optional
+  // Represents the total number of distinct scopes allowed per destination for the source reporting origin.
+  // This is used to calculate the information gain for event-level reports.
+  // Attribution scope limit values must be positive integers.  
+  // Defaults to null if omitted.
+  "attribution_scope_limit": <int>,
+
+  // Optional
+  // Represents a list of attribution scopes for a particular source.
+  // This is used to define the values that will be used to perform pre-attribution filtering.
+  // Attribution scope values must be strings. Each string has a maximum length of 50.
+  // The maximum length of the attribution_scopes list is the minimum of the attribution_scope_limit value and 20.
+  // Defaults to the empty list if omitted.
+  "attribution_scopes": <list of strings>,
+
+  // Optional
+  // Represents the maximum number of event states that an API caller plans to use across all
+  // subsequent event-source registrations.
+  // Example: default event source for event-level reports supports 1 attribution report, 1 reporting window,
+  // and 1 bit of trigger data for a max number of event states of 3.
+  // This is used to calculate the information gain for event-level reports.
+  // This is not required for navigation sources because navigations require explicit user action
```

Comment:
IMO this sentence is a bit ambiguous in that it could be read as meaning the field isn't required for *this* source if *this* source is navigation, but that's not the case.